onnxruntime/python/tools/transformers/fusion_embedlayer.py

```diff
@@ -126,41 +144,55 @@ def fuse(self, node, input_name_to_nodes, output_name_to_node):
         else:
             input_ids, input_ids_cast_node = self.utils.cast_input_to_int32(input_ids)
 
-        if self.model.find_graph_input(segment_ids):
-            casted, segment_ids = self.utils.cast_graph_input_to_int32(segment_ids)
-        else:
-            segment_ids, segment_ids_cast_node = self.utils.cast_input_to_int32(segment_ids)
-
-            # Cast might be removed by OnnxRuntime.
-            _, segment_id_path, _ = self.model.match_parent_paths(
-                segment_ids_cast_node,
-                [(['ConstantOfShape', 'Concat', 'Unsqueeze', 'Gather', 'Shape', 'Cast'], [0, 0, 1, 0, 0, 0]),
-                 (['ConstantOfShape', 'Concat', 'Unsqueeze', 'Gather', 'Shape'], [0, 0, 1, 0, 0])], output_name_to_node)
-
-            if segment_id_path and input_ids_cast_node and input_ids_cast_node.input[0] == segment_id_path[-1].input[0]:
-                logger.debug("Simplify semgent id path...")
-                self.model.add_node(
-                    helper.make_node('Shape', inputs=[input_ids_cast_node.input[0]], outputs=["input_shape"]))
-                self.model.add_node(
-                    helper.make_node('ConstantOfShape',
-                                     inputs=["input_shape"],
-                                     outputs=["zeros_for_input_shape"],
-                                     value=helper.make_tensor("value", onnx.TensorProto.INT32, [1], [1])))
-                segment_ids = "zeros_for_input_shape"
+        if is_distill == False:
```

Comment:
It seems that is_distill appears many times. Maybe it is better to implement it in a sparate function or class, insteading of mixing it with existing graph.

Reply:
refactored